### DISCUSSION TITLE
Is there a healtcheck function for the broker

### DISCUSSION THREAD

**Discussion by devraj** [taskiq-python/taskiq#112] [Q&A] [answered]
I maintain this [python stack](https://github.com/anomaly/lab-python-server) template for our projects and have recently moved to using `taskiq` (which I am enjoying greatly, thank you everyone for all the work).

It's largely ` FastAPI` + `SQLAlchemy` based infrastructure running in a container. The template provides `/healthcheck` endpoint for clusters to be able to sue to see if a node is healthy.

I was wondering if `taskiq` brokers have `healthcheck` that I can access, mostly to be able to report back on if the api endpoints are able to write to the queue.

**Comment by s3rius** [comment, marked as answer]
That's a really good suggestion. For now there's no easy way of doing so. 

Workers are expected to restart on errors, or they can self-heal from time to time. But on the client side, the easiest way would be to just check the queue by hand.

**Reply by s3rius** [reply]
I guess we can create a new function to check if the broker is up and running.

**Reply by devraj** [reply]
Thanks for your response. 

That's what I have been doing so far, checking if `redis` and `rabbitmq` are alive and deeming that to be `healthy`, if `taskiq` did provide a function then it would be more reliable, as the framework would be able to return a status as opposed to the individual components.
